frontend/src/PlayersView/PlayersView.jsx

```diff
@@ -1,28 +1,80 @@
-import StatsTable from "./StatsTable";
+import StatsTable from "../TableComponents/StatsTable";
+import SortMenu from "../TableComponents/SortMenu";
 import Header from "../Header/Header";
+import SearchBar from "../TableComponents/SearchBar";
 import { useState, useEffect } from "react";
+import TablePagination from "@mui/material/TablePagination";
 
 function PlayersView() {
   const [playersList, setPlayersList] = useState([]);
+  const [playersDisplayed, setPlayersDisplayed] = useState([]);
   const [page, setPage] = useState(1);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [sortType, setSortType] = useState("");
+  const [searchQuery, setSearchQuery] = useState("");
 
-  async function loadPlayers(page) {
-    const response = await fetch(
-      `https://nba-stats-db.herokuapp.com/api/playerdata/?page=${page}`
-    );
-    const data = await response.json();
-    const players = data.results;
+  const handleChangePage = (event, newPage) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  async function loadPlayers(sortType) {
+    let queryUrl = new URL("http://localhost:5000/players");
+    queryUrl.searchParams.append("sortType", sortType);
+    const response = await fetch(queryUrl);
+    const players = await response.json();
+    setPlayersList(players);
+  }
+
+  async function searchPlayer(playerName) {
+    if (playerName === "") {
+      return;
+    }
+
+    let queryUrl = new URL("http://localhost:5000/searchPlayers");
+    queryUrl.searchParams.append("playerName", playerName);
+    const response = await fetch(queryUrl);
+    const players = await response.json();
     setPlayersList(players);
   }
 
   useEffect(() => {
-    loadPlayers(page);
-  }, [page]);
+    const start = page * rowsPerPage;
+    const end = start + rowsPerPage;
+    setPlayersDisplayed(playersList.slice(start, end));
+  }, [page, rowsPerPage]);
+
```

Comment:
https://github.com/GloriaXM/numbers-game/assets/112780860/65ed1335-0791-435e-a86d-ae1c6139cd6e

Reply:
nit: would be much easier to view if you can widen the player name column so everything is 1-line

Reply:
I widened the column and used some css to handle exceptionally long names
![Screenshot 2024-07-09 at 11 48 49 AM](https://github.com/GloriaXM/numbers-game/assets/112780860/b38e2b2e-47e6-4063-b067-28bac9cf3d42)